### PR TITLE
Added badges to README/cleaned up

### PR DESCRIPTION
**Pull request by mdawsonuk**
Badges to determine current Master/Develop build statuses have been added, as well as supported .Net versions, branch staleness indicators, and other information.
The Contributing details has been moved CONTRIBUTING.md as per [Github's recommendations](https://help.github.com/en/articles/setting-guidelines-for-repository-contributors).

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+when an individual is representing the project or its community. Examples of
+representing a project or community include using an official project e-mail
+address, posting via an official social media account, or acting as an appointed
+representative at an online or offline event. Representation of a project may be
+further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at stuajnht@users.noreply.github.com. All
```
Comment (stuajnht) [inline CODE_OF_CONDUCT.md]: This is currently a GitHub no-reply address, which cannot accept email, so it is of no use for being able to report any problems.

The email address to be used will need to be thought about. I don't want to put down my personal one, and _CSSS_ does not currently have any registered domain name that can be used.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Contributing
+Thanks for your interest in contributing to this project. contribute or report issues in the following ways:
```
Comment (stuajnht) [inline CONTRIBUTING.md]: You're missing "_You can_" at the start of the second sentence.

File: README.md [hunk 3]
```diff
@@ -1,9 +1,48 @@
 # CyberSecurity Scoring System (CSSS)
 
+|Branch|Status (Windows Build)|Status (Linux Build)|Last Commit|
+|---|---|---|---|
+|Master|[![AppVeyor Build status](https://img.shields.io/appveyor/ci/stuajnht/CSSS?label=AppVeyor%20Build)](https://ci.appveyor.com/project/stuajnht/csss)|[![Build Status](https://img.shields.io/travis/stuajnht/CSSS.svg?label=Travis%20Build)](https://travis-ci.org/stuajnht/CSSS) |[![Last Master Commit](https://img.shields.io/github/last-commit/stuajnht/CSSS.svg)]()
```
Comment (stuajnht) [inline README.md]: Could you remove the rogue space please?

```
...(https://travis-ci.org/stuajnht/CSSS) |[![Last Master Commit]...
----------------------------------------^
```

File: README.md [hunk 4]
```diff
@@ -1,9 +1,48 @@
 # CyberSecurity Scoring System (CSSS)
 
+|Branch|Status (Windows Build)|Status (Linux Build)|Last Commit|
+|---|---|---|---|
+|Master|[![AppVeyor Build status](https://img.shields.io/appveyor/ci/stuajnht/CSSS?label=AppVeyor%20Build)](https://ci.appveyor.com/project/stuajnht/csss)|[![Build Status](https://img.shields.io/travis/stuajnht/CSSS.svg?label=Travis%20Build)](https://travis-ci.org/stuajnht/CSSS) |[![Last Master Commit](https://img.shields.io/github/last-commit/stuajnht/CSSS.svg)]()
+|Develop|[![AppVeyor Build status](https://img.shields.io/appveyor/ci/stuajnht/CSSS/develop?label=AppVeyor%20Build)](https://ci.appveyor.com/project/stuajnht/csss/develop)|[![Develop Build Status](https://img.shields.io/travis/stuajnht/CSSS/develop?label=Travis%20Build)]()|[![Last Develop Commit](https://img.shields.io/github/last-commit/stuajnht/CSSS/develop.svg)]()
```
Comment (stuajnht) [inline README.md]: [`https://ci.appveyor.com/project/stuajnht/csss/develop`](https://ci.appveyor.com/project/stuajnht/csss/develop) doesn't seem to link to anything?

File: README.md [hunk 5]
```diff
@@ -1,9 +1,48 @@
 # CyberSecurity Scoring System (CSSS)
 
+|Branch|Status (Windows Build)|Status (Linux Build)|Last Commit|
+|---|---|---|---|
+|Master|[![AppVeyor Build status](https://img.shields.io/appveyor/ci/stuajnht/CSSS?label=AppVeyor%20Build)](https://ci.appveyor.com/project/stuajnht/csss)|[![Build Status](https://img.shields.io/travis/stuajnht/CSSS.svg?label=Travis%20Build)](https://travis-ci.org/stuajnht/CSSS) |[![Last Master Commit](https://img.shields.io/github/last-commit/stuajnht/CSSS.svg)]()
+|Develop|[![AppVeyor Build status](https://img.shields.io/appveyor/ci/stuajnht/CSSS/develop?label=AppVeyor%20Build)](https://ci.appveyor.com/project/stuajnht/csss/develop)|[![Develop Build Status](https://img.shields.io/travis/stuajnht/CSSS/develop?label=Travis%20Build)]()|[![Last Develop Commit](https://img.shields.io/github/last-commit/stuajnht/CSSS/develop.svg)]()
+
+[![.Net Versions](https://img.shields.io/badge/.Net%20Versions-4.0|4.5|4.6|4.7-blue.svg)]()
```
Comment (stuajnht) [inline README.md]: I think .Net 4.8 should also be supported. Please check first before adding!

File: README.md [hunk 6]
```diff
@@ -1,9 +1,48 @@
 # CyberSecurity Scoring System (CSSS)
 
+|Branch|Status (Windows Build)|Status (Linux Build)|Last Commit|
+|---|---|---|---|
+|Master|[![AppVeyor Build status](https://img.shields.io/appveyor/ci/stuajnht/CSSS?label=AppVeyor%20Build)](https://ci.appveyor.com/project/stuajnht/csss)|[![Build Status](https://img.shields.io/travis/stuajnht/CSSS.svg?label=Travis%20Build)](https://travis-ci.org/stuajnht/CSSS) |[![Last Master Commit](https://img.shields.io/github/last-commit/stuajnht/CSSS.svg)]()
+|Develop|[![AppVeyor Build status](https://img.shields.io/appveyor/ci/stuajnht/CSSS/develop?label=AppVeyor%20Build)](https://ci.appveyor.com/project/stuajnht/csss/develop)|[![Develop Build Status](https://img.shields.io/travis/stuajnht/CSSS/develop?label=Travis%20Build)]()|[![Last Develop Commit](https://img.shields.io/github/last-commit/stuajnht/CSSS/develop.svg)]()
+
+[![.Net Versions](https://img.shields.io/badge/.Net%20Versions-4.0|4.5|4.6|4.7-blue.svg)]()
+[![Status](https://img.shields.io/badge/status-Pre%20Release-red.svg)]()
+[![Releases](https://img.shields.io/github/v/release/stuajnht/CSSS?include_prereleases)]()
```
Comment (stuajnht) [inline README.md:5]: Does this need to point to the GitHub releases page?

File: README.md [hunk 7]
```diff
@@ -1,9 +1,48 @@
...
+|Develop|[![AppVeyor Build status](https://img.shields.io/appveyor/ci/stuajnht/CSSS/develop?label=AppVeyor%20Build)](https://ci.appveyor.com/project/stuajnht/csss/develop)|[![Develop Build Status](https://img.shields.io/travis/stuajnht/CSSS/develop?label=Travis%20Build)]()|[![Last Develop Commit](https://img.shields.io/github/last-commit/stuajnht/CSSS/develop.svg)]()
+
+[![.Net Versions](https://img.shields.io/badge/.Net%20Versions-4.0|4.5|4.6|4.7-blue.svg)]()
+[![Status](https://img.shields.io/badge/status-Pre%20Release-red.svg)]()
+[![Releases](https://img.shields.io/github/v/release/stuajnht/CSSS?include_prereleases)]()
+[![Maintenance](https://img.shields.io/maintenance/yes/2019.svg)]()
+[![Known Vulnerabilities](https://snyk.io/test/github/stuajnht/CSSS/badge.svg?targetFile=CSSS.sln)](https://snyk.io/test/github/stuajnht/CSSS?targetFile=CSSS.sln)
+[![Contributors](https://img.shields.io/github/contributors/stuajnht/CSSS.svg)]()
```
Comment (stuajnht) [inline README.md:8]: Does this need to point to the GitHub contributors page?

File: README.md [hunk 8]
```diff
@@ -1,9 +1,48 @@
...
+[![Forks](https://img.shields.io/github/forks/stuajnht/CSSS.svg)]()
```
Comment (stuajnht) [inline README.md]: Does this need to point to the GitHub forks page?

File: README.md [hunk 9]
```diff
@@ -1,9 +1,48 @@
...
+[![Issues](https://img.shields.io/github/issues/stuajnht/CSSS.svg)](https://github.com/Stormy102/Lancer/issues)
```
Comment (stuajnht) [inline README.md]: Could this point to the _CSSS_ project instead please?

File: README.md [hunk 10]
```diff
@@ -1,9 +1,48 @@
...
+
 CyberSecurity Scoring System (CSSS) is a program to assist with training for the CyberPatriot / CyberCenturion competitions.
 
 While it is currently possible to create [your own practice images](https://www.uscyberpatriot.org/competition/training-materials/practice-images), there is only a scoring client for Windows computers. To assist with training competitors with Windows, Ubuntu and Debian Operating Systems, this program has been created to provide a scoring system for them.
```
Comment (stuajnht) [inline README.md:15]: Could the build status table go after the introductory paragraphs, rather than at the top of the readme, please? The status badges can stay where they are.

File: README.md [hunk 11]
```diff
@@ -1,9 +1,48 @@
...
+
+### Version 0.4
+* Create scripts to install CSSS files into the correct locations (WinNT: root of `%SystemDrive%`, Linux: `/opt` directory)
+* Install additional software when CSSS is being installed to make sure all items are available for it to use
+* Create build scripts that install software that can be used when testing the checker system engine
+
+### Versions 0.5 onward
+* Create more check template files and relevant classes for the checker system engine to use (a new version for each check "category")
+* Create some example checker files with instructions to create images based around them with pre-made questions and scenarios
+
```
Comment (stuajnht) [inline README.md]: Could the disclaimer and To-Do sections go back to their original locations please? It makes more sense for users to know how to run the program than what it currently can't do.

File: README.md [hunk 12]
```diff
@@ -69,23 +108,7 @@ When you are ready to release the image, __with administrative privileges__ (e.g
...
-This project uses [*EditorConfig*](https://editorconfig.org/) to maintain a consistent coding style, using the [default EditorConfig file](https://docs.microsoft.com/en-gb/visualstudio/ide/editorconfig-code-style-settings-reference#example-editorconfig-file) that comes with Visual Studio (with [a few additional tweaks](.editorconfig)).
-
-If you are planning to contribute code, please make sure that you have [performed a code cleanup](https://docs.microsoft.com/en-us/visualstudio/ide/code-styles-and-code-cleanup?#apply-code-styles). Alternatively, install an extension to do this for you:
-
-* [Format document on Save](https://marketplace.visualstudio.com/items?itemName=mynkow.FormatdocumentonSave)
-* [Code Cleanup On Save](https://marketplace.visualstudio.com/items?itemName=MadsKristensen.CodeCleanupOnSave)
-
-## License Terms
-CSSS is publised under the GNU GPL v3 License, see the [LICENSE](LICENSE.md) file for more information.
+Thanks for your interest in contributing to this project. You can read more about contributing to the project [here](CONTRIBUTING.md)
```
Comment (stuajnht) [inline README.md]: Could this instead read:
```
Thanks for your interest in CSSS. Read more about [contributing to the project](CONTRIBUTING.md) to get started.
```

**Review by stuajnht — CHANGES_REQUESTED** [review]
Hi @Stormy102, thanks for this PR. I've made a few comments - could you address them please?

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (mdawsonuk) [reply CODE_OF_CONDUCT.md]: I have removed the email at this point - however, further consideration should be given into more private ways to contact the project team.

File: README.md (hunk 4 above)
Comment (mdawsonuk) [reply README.md]: Now correctly links to [https://ci.appveyor.com/project/stuajnht/csss/history?branch=develop](https://ci.appveyor.com/project/stuajnht/csss/history?branch=develop)

File: README.md (hunk 8 above)
Comment (mdawsonuk) [reply README.md]: Now points to GitHub forks page

File: README.md (hunk 9 above)
Comment (mdawsonuk) [reply README.md]: Fixed - the mistake of copy/pasting!

File: README.md (hunk 5 above)
Comment (mdawsonuk) [reply README.md]: Installed, built & ran tests - can confirm that it appears to work without issue. I'll add it as supported but potentially further work to completely verify testing could be done in a VM.

**Comment by mdawsonuk** [comment]
Made all of the necessary changes - hopefully everything is satisfactory now

File: CODE_OF_CONDUCT.md [hunk 13]
```diff
@@ -55,11 +55,11 @@ further defined and clarified by project maintainers.
...
-reported by contacting the project team at stuajnht@users.noreply.github.com. All
-complaints will be reviewed and investigated and will result in a response that
-is deemed necessary and appropriate to the circumstances. The project team is
-obligated to maintain confidentiality with regard to the reporter of an incident.
-Further details of specific enforcement policies may be posted separately.
+reported by contacting the project team. All complaints will be reviewed and
+investigated and will result in a response that is deemed necessary and
+appropriate to the circumstances. The project team is obligated to maintain
+confidentiality with regard to the reporter of an incident. Further details
+of specific enforcement policies may be posted separately.
```
Comment (stuajnht) [inline CODE_OF_CONDUCT.md]: I don't think the email address can be removed, as it says on the [Contributor Covenant](https://www.contributor-covenant.org) homepage:

> **Important!** You must add a contact method to the placeholder in the document so that people know how to report violations.

I am happy for the email address [projects-csss-contact@stuajnht.co.uk](projects-csss-contact@stuajnht.co.uk) to be used. Please revert this change and add in the email address.

File: README.md [hunk 14]
```diff
@@ -1,67 +1,23 @@
...
+[![.Net Versions](https://img.shields.io/badge/.Net%20Versions-4.0|4.5|4.6|4.7|4.8-blue.svg)]()
 [![Status](https://img.shields.io/badge/status-Pre%20Release-red.svg)]()
 [![Releases](https://img.shields.io/github/v/release/stuajnht/CSSS?include_prereleases)]()
 [![Maintenance](https://img.shields.io/maintenance/yes/2019.svg)]()
 [![Known Vulnerabilities](https://snyk.io/test/github/stuajnht/CSSS/badge.svg?targetFile=CSSS.sln)](https://snyk.io/test/github/stuajnht/CSSS?targetFile=CSSS.sln)
 [![Contributors](https://img.shields.io/github/contributors/stuajnht/CSSS.svg)]()
-[![Forks](https://img.shields.io/github/forks/stuajnht/CSSS.svg)]()
-[![Issues](https://img.shields.io/github/issues/stuajnht/CSSS.svg)](https://github.com/Stormy102/Lancer/issues)
+[![Forks](https://img.shields.io/github/forks/stuajnht/CSSS.svg)](https://github.com/stuajnht/CSSS/network/members)
+[![Issues](https://img.shields.io/github/issues/stuajnht/CSSS.svg)](https://github.com/stuajnht/CSSS/issues)
```
Comment (stuajnht) [inline README.md:11]: If we're adding a Code of Conduct, could you also add a badge for it please?

```
[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-v1.4%20adopted-ff69b4.svg)](code-of-conduct.md) 
```

File: README.md [hunk 15]
```diff
@@ -1,67 +1,23 @@
...
-    * Ubuntu 16.04
-    * Ubuntu 18.04
-    * Debian 9
-
-Other operating systems should work, but they are not supported as the CyberPatriot / CyberCenturion competitions do not use them.
+## Build Info
+|Branch|Status (Windows Build)|Status (Linux Build)|Last Commit|
+|---|---|---|---|
+|Master|[![AppVeyor Build status](https://img.shields.io/appveyor/ci/stuajnht/CSSS?label=AppVeyor%20Build)](https://ci.appveyor.com/project/stuajnht/csss)|[![Build Status](https://img.shields.io/travis/stuajnht/CSSS.svg?label=Travis%20Build)](https://travis-ci.org/stuajnht/CSSS)|[![Last Master Commit](https://img.shields.io/github/last-commit/stuajnht/CSSS.svg)]()
+|Develop|[![AppVeyor Build status](https://img.shields.io/appveyor/ci/stuajnht/CSSS/develop?label=AppVeyor%20Build)](https://ci.appveyor.com/project/stuajnht/csss/history?branch=develop)|[![Develop Build Status](https://img.shields.io/travis/stuajnht/CSSS/develop?label=Travis%20Build)]()|[![Last Develop Commit](https://img.shields.io/github/last-commit/stuajnht/CSSS/develop.svg)]()
```
Comment (stuajnht) [inline README.md]: Is there a way to filter the Travis build status to just the develop branch? I don't think there is, as I can't find any documentation about it. If you can, great! If not, it can stay as it is.

File: README.md [hunk 16]
```diff
@@ -1,67 +1,23 @@
...
 [![Known Vulnerabilities](https://snyk.io/test/github/stuajnht/CSSS/badge.svg?targetFile=CSSS.sln)](https://snyk.io/test/github/stuajnht/CSSS?targetFile=CSSS.sln)
 [![Contributors](https://img.shields.io/github/contributors/stuajnht/CSSS.svg)]()
-[![Forks](https://img.shields.io/github/forks/stuajnht/CSSS.svg)]()
-[![Issues](https://img.shields.io/github/issues/stuajnht/CSSS.svg)](https://github.com/Stormy102/Lancer/issues)
+[![Forks](https://img.shields.io/github/forks/stuajnht/CSSS.svg)](https://github.com/stuajnht/CSSS/network/members)
+[![Issues](https://img.shields.io/github/issues/stuajnht/CSSS.svg)](https://github.com/stuajnht/CSSS/issues)
 
 CyberSecurity Scoring System (CSSS) is a program to assist with training for the CyberPatriot / CyberCenturion competitions.
 
 While it is currently possible to create [your own practice images](https://www.uscyberpatriot.org/competition/training-materials/practice-images), there is only a scoring client for Windows computers. To assist with training competitors with Windows, Ubuntu and Debian Operating Systems, this program has been created to provide a scoring system for them.
```
Comment (stuajnht) [inline README.md:15]: Could you add a table of contents to the readme please, and re-order / move the sections to be:

```
<details>
  <summary>Contents</summary>

  * [Build Info](#build-info)
  * [Requirements](#requirements)
  * [Installing](#installing)
  * [Running](#running)
  * [Contributing](#contributing) (move To-Do sections to the contributing page)
  * [License Terms](#license-terms)
    * [NuGet Packages](#nuget-packages)
    * [Other Projects](#other-projects)
  * [Copyright](#copyright)
  * [Disclaimer](#disclaimer)
</details>
```

File: README.md [hunk 17]
```diff
@@ -108,7 +64,7 @@ When you are ready to release the image, __with administrative privileges__ (e.g
 > :warning: Running CSSS with the `-p` argument will encrypt the issue files. Make sure that you have a snapshot of your image before you run this command, otherwise you will need to set everything up again should there be any problems.
 
 ## Contributing
-Thanks for your interest in contributing to this project. You can read more about contributing to the project [here](CONTRIBUTING.md)
+Thanks for your interest in CSSS. Read more about [contributing to the project](CONTRIBUTING.md) to get started.
```
Comment (stuajnht) [inline README.md:106]: Could you add in the following please, and link to the Code of Conduct:

```
Please note that this project is released with a [Contributor Code of Conduct](#CODE_OF_CONDUCT.md). By participating in this project you agree to abide by its terms.
```

File: CONTRIBUTING.md [hunk 18]
```diff
@@ -1,5 +1,5 @@
 # Contributing
-Thanks for your interest in contributing to this project. contribute or report issues in the following ways:
+Thanks for your interest in contributing to this project. You can contribute or report issues in the following ways:
```
Comment (stuajnht) [inline CONTRIBUTING.md]: Please add a paragraph under the `Contributing` title with the quote:

```
> Please note that this project is released with a [Contributor Code of Conduct](#CODE_OF_CONDUCT.md). By participating in this project you agree to abide by its terms.
```

**Review by stuajnht — CHANGES_REQUESTED** [review]
Thanks for making the changes @Stormy102 - I've added just a few more change that I'd like made please. Once I've checked them, then I think we should be good to merge this PR.

File: README.md (hunk 15 above)
Comment (mdawsonuk) [reply README.md]: I can't find any documentation either. As a work-around, clicking on the badge will take you to the branches page, which will give an indicator of the latest builds on active branches.

File: CONTRIBUTING.md [hunk 19]
```diff
@@ -1,4 +1,5 @@
 # Contributing
+> Please note that this project is released with a [Contributor Code of Conduct](#CODE_OF_CONDUCT.md). By participating in this project you agree to abide by its terms.
 Thanks for your interest in contributing to this project. You can contribute or report issues in the following ways:
```
Comment (stuajnht) [inline CONTRIBUTING.md]: Could you add a newline between both of these? Currently the intro text is being combined with the notice when [viewing on GitHub](https://github.com/Stormy102/CSSS/blob/feature/ReadMeBadgify/CONTRIBUTING.md#contributing).

File: README.md [hunk 20]
```diff
@@ -1,140 +1,126 @@
...
+The required arguments have been chosen to spell 'COPS', as competitors are 'policing' the security of the computer.
+
+Under normal image building, the `-o` option should be passed, as this allows CSSS to run as it would for training, but without affecting the files used to list the 'issues'.
+
+When you are ready to release the image, __with administrative privileges__ (e.g. `sudo`, "Run as administrator") run CSSS with the `-p` option to prepare the necessary files and allow CSSS to start automatically on computer reboots.
+> :warning: Running CSSS with the `-p` argument will encrypt the issue files. Make sure that you have a snapshot of your image before you run this command, otherwise you will need to set everything up again should there be any problems.
+
+## Contributing
+Thanks for your interest in CSSS. Read more about [contributing to the project](CONTRIBUTING.md) to get started.
+Please note that this project is released with a [Contributor Code of Conduct](#CODE_OF_CONDUCT.md). By participating in this project you agree to abide by its terms.
```
Comment (stuajnht) [inline README.md]: Could you add a newline between the two paragraphs please?
```
... to get started.

Please note that...
```

File: README.md [hunk 21]
```diff
@@ -1,140 +1,126 @@
...
+## Copyright
+CyberPatriot is copyright of US Air Force Association
+
+CyberCentrurion is copyright of Cyber Security Challenge UK
+
+## Disclaimer
+This program is designed to score the state of a computer image against a pre-defined checklist of "issues". It is not designed to rate your computer security, provide security advice, or anything else that isn't related to the CyberPatriot / CyberCenturion competitions.
+
+## License Terms
+CSSS is publised under the GNU GPL v3 License, see the [LICENSE](LICENSE.md) file for more information.
```
Comment (stuajnht) [inline README.md]: License information needs to be moved after "Contributing" / before "NuGet Packages" as per the table of contents.

**Review by stuajnht — CHANGES_REQUESTED** [review]
Thanks again @Stormy102 - just a few spacing / layout changes and I should be happy to merge!

File: README.md [hunk 22]
```diff
@@ -1,138 +1,126 @@
...
-## Disclaimer
-This program is designed to score the state of a computer image against a pre-defined checklist of "issues". It is not designed to rate your computer security, provide security advice, or anything else that isn't related to the CyberPatriot / CyberCenturion competitions.
+# CyberSecurity Scoring System (CSSS)
+
+[![.Net Versions](https://img.shields.io/badge/.Net%20Versions-4.0|4.5|4.6|4.7|4.8-blue.svg)]()
+[![Status](https://img.shields.io/badge/status-Pre%20Release-red.svg)]()
+[![Releases](https://img.shields.io/github/v/release/stuajnht/CSSS?include_prereleases)]()
+[![Maintenance](https://img.shields.io/maintenance/yes/2019.svg)]()
+[![Contributor Covenant](https://img.shields.io/badge/Contributor%20Covenant-v1.4%20adopted-ff69b4.svg)](CODE_OF_CONDUCT.md)
+[![Known Vulnerabilities](https://snyk.io/test/github/stuajnht/CSSS/badge.svg?targetFile=CSSS.sln)](https://snyk.io/test/github/stuajnht/CSSS?targetFile=CSSS.sln)
```
Comment (stuajnht) [inline README.md]: Could you have a look to see if you can get a badge to display for Snyk, as it currently just says "Known Vulnerabilities" when [viewed on GitHub](https://github.com/Stormy102/CSSS/blob/feature/ReadMeBadgify/README.md#cybersecurity-scoring-system-csss). I think it may only work for a [Node.js, Ruby or Java GitHub repository](https://snyk.io/docs/badges/#github-badge).

Comment (mdawsonuk) [reply README.md]: You need to setup Snyk correctly for it to display. I'm not sure if you've done this yet, so I'll comment out the badge for the future.

File: CONTRIBUTING.md [hunk 23]
```diff
@@ -0,0 +1,39 @@
+# Contributing
+> Please note that this project is released with a [Contributor Code of Conduct](#CODE_OF_CONDUCT.md). By participating in this project you agree to abide by its terms.
```
Comment (stuajnht) [inline CONTRIBUTING.md]: Sorry, this is my fault. Could you remove the `#` from the link, so that it points to the file, rather than trying to find an anchor on the current page?

File: README.md [hunk 24]
```diff
@@ -1,138 +1,126 @@
...
+
+Under normal image building, the `-o` option should be passed, as this allows CSSS to run as it would for training, but without affecting the files used to list the 'issues'.
+
+When you are ready to release the image, __with administrative privileges__ (e.g. `sudo`, "Run as administrator") run CSSS with the `-p` option to prepare the necessary files and allow CSSS to start automatically on computer reboots.
+> :warning: Running CSSS with the `-p` argument will encrypt the issue files. Make sure that you have a snapshot of your image before you run this command, otherwise you will need to set everything up again should there be any problems.
+
+## Contributing
+Thanks for your interest in CSSS. Read more about [contributing to the project](CONTRIBUTING.md) to get started.
+
+Please note that this project is released with a [Contributor Code of Conduct](#CODE_OF_CONDUCT.md). By participating in this project you agree to abide by its terms.
```
Comment (stuajnht) [inline README.md]: As before, this is my fault. Could you remove the `#` from the link, so that it points to the file, rather than trying to find an anchor on the current page?

**Review by stuajnht — CHANGES_REQUESTED** [review]
Hi @Stormy102 - sorry, both changes needed are my fault. I didn't check the text that I wanted you to use properly before I posted it. Do you mind just removing the `#` as needed?

File: README.md (hunk 22 above)
Comment (stuajnht) [reply README.md]: I've added CSSS to my Snyk account. However, it looks like you can't just add a badge direct to the `CSSS.sln` file, but for each project inside it?

For example, Snyk gives me a badge code of:
```  
[![Known Vulnerabilities](https://snyk.io/test/github/stuajnht/CSSS/badge.svg?targetFile=CSSS/CSSS.csproj)](https://snyk.io/test/github/stuajnht/CSSS?targetFile=CSSS/CSSS.csproj)
```

... which renders as:
[Known Vulnerabilities](https://snyk.io/test/github/stuajnht/CSSS?targetFile=CSSS/CSSS.csproj)

... meaning that we'd need to have one badge for each project in the solution (which will start to look messy).

However, as mentioned in [my previous comment](https://github.com/stuajnht/CSSS/pull/3#discussion_r338809991), I don't think Snyk badges work for .Net projects (at the moment?). Therefore, it is worth just removing the commented out badge? Unless you can find another way to get it to work?

Comment (mdawsonuk) [reply README.md]: Snyk does have support for .Net projects - however, only the Command Line is supported for .sln - see [here](https://snyk.io/docs/snyk-for-dotnet/)

A work-around would be displaying the badge for the whole repo. However, until the master branch (or the branch you specified for Snyk to monitor) is updated, the badge won't display properly. Therefore I'm going to remove the badge for now.

**Comment by Stormy102** [comment]
I'll fix the merge conflicts on my branch and commit

**Review by stuajnht — APPROVED** [review]

**Comment by stuajnht** [comment]
Thanks @Stormy102 for adding the `Code of Conduct` and `Contributing` documents, and for making the (many) changes!
